web/gds-admin-ui/src/pages/app/details/BasicDetails/components/Ivms101RecordForm.js

```diff
@@ -0,0 +1,166 @@
+import CountryOptions from 'components/CountryOptions';
+import Field from 'components/Field';
+import { ModalCloseButton, ModalContext } from 'components/Modal';
+import NationalIdentifierOptions from 'components/NationalIdentifierOptions';
+import useSafeDispatch from 'hooks/useSafeDispatch';
+import React from 'react'
+import { Alert, Button, Form, FormGroup } from 'react-bootstrap'
+import { useForm } from 'react-hook-form';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { getIvmsErrorState, getVaspDetailsLoadingState } from 'redux/selectors';
+import { clearIvms101ErrorMessage, updateIvms101Response } from 'redux/vasp-details';
+import { getIvms101RecordInitialValues } from 'utils/form-references';
+import AddressesFieldArray from './AddressesFieldArray';
+import NameIdentifiersFieldArray from './NameIdentifiersFieldArray';
+
+function Ivms101RecordForm({ data }) {
+    const { register, control, handleSubmit } = useForm({
+        defaultValues: getIvms101RecordInitialValues(data)
+    })
+    const params = useParams()
+    const [, setIsOpen] = React.useContext(ModalContext)
+    const nameIdentifiersFieldArrayRef = React.useRef()
+    const localNameIdentifiersFieldArrayRef = React.useRef()
+    const phoneticNameIdentifiersFieldArrayRef = React.useRef()
+    const dispatch = useDispatch()
+    const safeDispatch = useSafeDispatch(dispatch)
+    const ivmsErrorState = useSelector(getIvmsErrorState)
+    const isLoading = useSelector(getVaspDetailsLoadingState)
+
+    React.useEffect(() => {
+        const timeout = setTimeout(() => {
+            safeDispatch(clearIvms101ErrorMessage())
+        }, 30000)
```

Comment:
I noticed that if I closed the modal and then opened it up again, if it was still within this 30 second window the error would still be there even though the form has been reset.